src/mp4/ilst/read.rs

```diff
@@ -187,17 +202,32 @@ where
 		// We don't care about the locale
 		reader.seek(SeekFrom::Current(4))?;
 
-		let content_len = (data_atom.len - 16) as usize;
-		if content_len == 0 {
-			// We won't add empty atoms
-			return Ok(None);
```

Comment:
This early return was wrong. The comment states that multiple "data" atoms could occur, but the function aborts on the first empty atom.